contracts/strategy/aave/AaveV3Lender.sol

```diff
@@ -0,0 +1,387 @@
+        // Get the bit map data config.
+        uint256 data = lendingPool
+            .getReserveData(address(asset))
+            .configuration
+            .data;
+        // Get out the supply cap for the asset.
+        uint256 cap = (data & ~SUPPLY_CAP_MASK) >>
+            SUPPLY_CAP_START_BIT_POSITION;
+        // Adjust to the correct decimals.
+        return cap * (10 ** decimals);
+    }
+
+    /**
+     * @notice Gets the max amount of `asset` that can be withdrawn.
+     * @dev Defaults to an unlimited amount for any address. But can
+     * be overridden by strategists.
+     *
+     * This function will be called before any withdraw or redeem to enforce
+     * any limits desired by the strategist. This can be used for illiquid
+     * or sandwichable strategies. It should never be lower than `totalIdle`.
+     *
+     *   EX:
+     *       return TokenIzedStrategy.totalIdle();
+     *
+     * This does not need to take into account the `_owner`'s share balance
+     * or conversion rates from shares to assets.
+     *
+     * @param . The address that is withdrawing from the strategy.
+     * @return . The available amount that can be withdrawn in terms of `asset`
+     */
+    function availableWithdrawLimit(
+        address /*_owner*/
+    ) public view override returns (uint256) {
+        return TokenizedStrategy.totalIdle() + asset.balanceOf(address(aToken));
+    }
+
+    /**
+     * @notice Allows `management` to manually swap a token the strategy holds.
+     * @dev This can be used if the rewards controller has since removed a reward
+     * token so the normal harvest flow doesn't work, for retroactive airdrops.
+     * or just to slowly sell tokens at specific times rather than during harvests.
+     *
+     * @param _token The address of the token to sell.
+     * @param _amount The amount of `_token` to sell.
+     * @param _minAmountOut The minimum of `asset` to get out.
+     */
+    function sellRewardManually(
+        address _token,
+        uint256 _amount,
+        uint256 _minAmountOut
+    ) external onlyManagement {
+        _swapFrom(
+            _token,
+            address(asset),
+            Math.min(_amount, ERC20(_token).balanceOf(address(this))),
+            _minAmountOut
+        );
+    }
+
+    /**
+     * @notice Set the `minAmountToSellMapping` for a specific `_token`.
+     * @dev This can be used by management to adjust wether or not the
+     * _claimAndSellRewards() function will attempt to sell a specific
+     * reward token. This can be used if liquidity is to low, amounts
+     * are to low or any other reason that may cause reverts.
+     *
+     * @param _token The address of the token to adjust.
+     * @param _amount Min required amount to sell.
+     */
+    function setMinAmountToSellMapping(
+        address _token,
+        uint256 _amount
+    ) external onlyManagement {
+        minAmountToSellMapping[_token] = _amount;
```

Comment:
lack of validation here

Reply:
Other than checking if _amount is lower than the type(uint256).max what do you propose to be validated here?